components/script/dom/idbopendbrequest.rs

```diff
@@ -0,0 +1,378 @@
+
+        let trusted_request = Trusted::new(self);
+        let name = name.to_string();
+        ROUTER.add_route(
+            receiver.to_opaque(),
+            Box::new(move |message| {
+                let global = trusted_request.root().global();
+                let trusted_request = trusted_request.clone();
+                let response_listener = response_listener.clone();
+                let name = name.clone();
+
+                task_source.queue(
+                    task!(set_request_result_to_database: move || {
+                        let (result, did_upgrade) =
+                            response_listener.handle_open_db(name, version, message.to().unwrap());
+                        // If an upgrade event was created, it will be responsible for
+                        // dispatching the success event
+                        if !did_upgrade {
+                            let request = trusted_request.root();
+                            let global = request.global();
+                            match result {
+                                Ok(db) => {
+                                    request.dispatch_success(&*db);
+                                },
+                                Err(dom_exception) => {
+                                    request.set_result(HandleValue::undefined());
+                                    request.set_error(dom_exception);
+                                    let event = Event::new(
+                                        &global,
+                                        Atom::from("error"),
+                                        EventBubbles::Bubbles,
+                                        EventCancelable::Cancelable,
+                                    );
+                                    event.upcast::<Event>().fire(request.upcast());
+                                }
+                            }
+                        }
+                    }),
+                    global.upcast(),
+                ).unwrap();
+            }),
+        );
+
+        global
+            .resource_threads()
+            .sender()
+            .send(IndexedDBThreadMsg::Sync(open_operation))
+            .unwrap();
+    }
+
+    pub fn delete_database(&self, name: String) {
+        let global = self.global();
+
+        let (sender, receiver) = ipc::channel(global.time_profiler_chan().clone()).unwrap();
+        let task_source = global.database_access_task_source();
+        let response_listener = OpenRequestListener {
+            open_request: Trusted::new(self),
+        };
+
+        let delete_operation =
+            SyncOperation::DeleteDatabase(sender, global.origin().immutable().clone(), name);
+
+        let trusted_global = Trusted::new(&*global);
+        ROUTER.add_route(
+            receiver.to_opaque(),
+            Box::new(move |message| {
+                let response_listener = response_listener.clone();
+                let global = trusted_global.root();
```

Comment:
Same issue here - we need to have a task canceller already instead of rooting this global.